core-bundle/src/Security/Authentication/RememberMe/ExpiringTokenBasedRememberMeServices.php

```diff
@@ -182,6 +182,6 @@ private function createRememberMeCookie(Request $request, string $series, string
 
     private function encodeSeries(string $series): string
     {
-        return hash_hmac('sha256', $series, $this->secret);
+        return hash_hmac('sha256', $series, $this->secret, true);
     }
```

Comment:
We would have to do the encoding of the cookie ourselfs if we want to directly work with the binary values

```php
    protected function decodeCookie($rawCookie)
    {
        return array_map('base64_decode', explode('-', $rawCookie));
    }
    protected function encodeCookie(array $cookieParts)
    {
        return implode('-', array_map('base64_encode', $cookieParts));
    }
```

Reply:
Added in fc54f7bcb462656f1042b2ad01fc1d920cd60208. Why did you decide to use `-` instead of `AbstractRememberMeServices::COOKIE_DELIMITER` (which is `:`)?

Reply:
You're aware this implementation differs from the well-used Symfony one? We are no longer encoding the final value, whereas Sf does that: https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/Security/Http/RememberMe/AbstractRememberMeServices.php#L254
I really don't see the point in changing the original approach…

Reply:
It is unnecessary to encode the final value, because all parts are already encoded.

> I really don't see the point in changing the original approach…

See the original issue description.